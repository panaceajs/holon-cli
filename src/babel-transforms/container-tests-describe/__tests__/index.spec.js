const pluginTester = require('babel-plugin-tester');
const containerDefaultExport = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: containerDefaultExport,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/with-default-export.js',
      snapshot: false
    },
    {
      title: 'it should add default export.',
      fixture: '__fixtures__/without-default-export.js'
    }
  ]
});
