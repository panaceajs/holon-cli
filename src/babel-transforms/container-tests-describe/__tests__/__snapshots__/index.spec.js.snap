// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-container-tests-describe it should add default export.: it should add default export. 1`] = `
"
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

console.log('nothing going on here');
describe('\`undefined\` container', () => {});
"
`;
